scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+
+                }
+            }
+            commitIfNeeded(channel);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public boolean waitUntil(String originator, TaskId taskid, String channel, String key, String predicate)
+            throws InvalidChannelException, CompilationException {
+        Channel chosenChannel = getChannel(channel);
+        Serializable value = chosenChannel.get(key);
+        try {
+            return evaluateClosure(predicate, BiPredicate.class).test(key, value);
+        } catch (CompilationException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public PredicateActionResult waitUntilThen(String originator, TaskId taskid, String channel, String key,
+            String predicate, String thenRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
+        PredicateActionResult answer;
+        try {
+            Channel chosenChannel = getChannel(channel);
+            Serializable value = chosenChannel.get(key);
+            if (evaluateClosure(predicate, BiPredicate.class).test(key, value)) {
+                answer = new PredicateActionResult(true,
+                                                   chosenChannel.compute(key,
+                                                                         (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(thenRemappingFunction,
+                                                                                                                                                                    BiFunction.class)));
+            } else {
+                throw new IllegalStateException("Predicate " + predicate + " is not met");
+            }
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run waitUntil" + ON_KEY + QUOTE + key + QUOTE + RESULT_IS + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    private void logWithContext(TaskId taskId, String channel, String message, Level level)
+            throws InvalidChannelException {
+        String channelHeader = "[Synchronization]" + (channel != null ? "[Channel=" + channel + "] " : "");
```

Comment:
Can we replace `Synchronization` by `Synchronization.class.getSimpleName()`?

Same remark for `Channel`

Reply:
ok